core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -49,6 +49,16 @@ public class PartitionsTable extends BaseMetadataTable {
         new Schema(
             Types.NestedField.required(1, "partition", Partitioning.partitionType(table)),
             Types.NestedField.required(4, "spec_id", Types.IntegerType.get()),
+            Types.NestedField.required(
```

Comment:
Question: did we consider making it optional, and returning null?  A precedent may be snapshots table parentId field.

Reply:
thank you Szehon, that make sense and I updated to mark both optional and leave its default to null like parent_id in snapshots metadata table